### DISCUSSION TITLE
Alignment

### DISCUSSION THREAD

**Discussion by vid1988** [bookfere/Ebook-Translator-Calibre-Plugin#308] [Q&A] [answered]
Hi, 

i used your tool to translate an epub via chatgpt. It states, the some paragraphs are non-aligned, and in case i try to output in the advanced translation GUI, it asks, if i really want to write it without checking the alignment. 
I also used the merge paragraphs feature, I guess it has something to do with that?

How do I check and correct the alignment? 

br
David

**Comment by bookfere** [comment, marked as answer]
When you enable the merge translation feature, the plugin needs to ensure that the original text and the translation are aligned by keeping the line breaks in the same positions. However, GenAI often doesn't respect the original line breaks, causing them to be missing in the translation.

Non-aligned items can be identified by their background color and filtered using the filter feature in Advanced mode. You can use the Review feature to manually fix any misalignments in the translation, or you can retranslate in the hope of achieving an ideal translation.

You can also ignore the alert dialog and output the ebook without alignment checking. However, this will result in non-aligned translations being appended to the corresponding merge units as a whole, making it difficult to read the original text and its translation paragraph by paragraph.

**Comment by vid1988** [comment]
So "alignment" means number of real paragraphs per "translated paragraphs" must fit in original language to translated language. 
The Tool just kept the original version in case of non-alignment (i normally completely replaced the original). 

Thank you so much, issue resolved :-)
